pyanaconda/ui/tui/spokes/network.py

```diff
@@ -51,16 +54,29 @@ def __init__(self, app, data, storage, payload, instclass):
         self.errors = []
 
     def initialize(self):
-        for name in nm.nm_devices():
+        self._load_new_devices()
+
+        EditTUISpoke.initialize(self)
+
+    def _load_new_devices(self):
+        """ Register new devices to supported devices list. """
+        devices = nm.nm_devices()
+        intf_dumped = network.dumpMissingDefaultIfcfgs()
+        if intf_dumped:
+            log.debug("Dumped interfaces: {0}".format(intf_dumped))
+
+        for name in devices:
+            if name in self.supported_devices:
+                # this device is already registered
+                continue
             if nm.nm_device_type_is_ethernet(name):
                 # ignore slaves
                 if nm.nm_device_setting_value(name, "connection", "slave-type"):
                     continue
                 self.supported_devices.append(name)
 
-        EditTUISpoke.initialize(self)
         if not self.data.network.seen:
-            self._update_network_data()
+            self._update_network_data(devices)
```

Comment:
I think we can keep doing _update_network_data() only in initialize and apply (not in every _load_new_devices / refresh)? So I'd keep it in initialize after _load_new_devices().